internal/controller/agenttags/tags.go

```diff
@@ -42,6 +42,16 @@ func SetTag(tags []string, key, val string) []string {
 	return tags
 }
 
+func RemoveTag(tags []string, key string) []string {
+	i := slices.IndexFunc(tags, func(tag string) bool {
+		return strings.HasPrefix(tag, key+"=")
+	})
+	if i < 0 {
+		return tags
+	}
+	return slices.Delete(tags, i, i+1)
```

Comment:
Avoids dealing with an index `i`:

```suggestion
	return slices.DeleteFunc(tags, func(tag string) bool {
		return strings.HasPrefix(tag, key+"=")
	})
```